### PR TITLE
Group management

### PR DESCRIPTION
**Pull request by dynamiccast**
Add group
Edit group name
Add members to group
Add application to group
Remove users from a group
Remove applications from a group
Admin can list all applications
Users can list applications only if they belongs to groups they also belong to
Apps model is rename App to follow the same Model name convention
Rename former `AppController` (used for static file) into `NanocloudController`

### CODE REVIEW

File: api/controllers/AppController.js [hunk 1]
```diff
@@ -19,46 +19,91 @@
...
+
+        return resolve(apps);
+      });
     });
+  },
+
+  find(req, res) {
+
+    this._getApps(req.user)
+      .then((apps) => {
```
Comment (willrnch) [inline api/controllers/AppController.js:74]: handle error case

File: api/controllers/AppController.js [hunk 2]
```diff
@@ -19,46 +19,91 @@
...
+                password: machine.password,
+                "remote-app": '',
+                protocol: 'rdp',
+                "app-name": app.alias
+              });
+            });
+
+            return res.ok(connections);
+          });
+      });
```
Comment (willrnch) [inline api/controllers/AppController.js:113]: handle error case

File: api/controllers/GroupController.js [hunk 3]
```diff
@@ -25,7 +25,15 @@
...
 module.exports = {
-	
-};
 
+  findOne: function(req, res) {
+
+    Group.findOne(req.allParams()["id"])
+      .populate("members")
+      .populate('apps')
+      .then(res.ok);
```
Comment (willrnch) [inline api/controllers/GroupController.js]: handle error case

File: api/controllers/UserController.js [hunk 4]
```diff
@@ -37,6 +37,8 @@ module.exports = {
       return res.send(me);
     }
 
-    return JsonApiService.findRecords(req, res);
+    return User.find()
+      .populate("groups")
+      .then((res.ok));
```
Comment (willrnch) [inline api/controllers/UserController.js]: handle error case

File: api/models/Group.js [hunk 5]
```diff
@@ -25,10 +25,34 @@
...
 
+  autoPK: false,
+
   attributes: {
+    id: {
+      type: 'string',
+      primaryKey: true,
+      unique: true,
+      index: true,
+      uuidv4: true,
```
Comment (willrnch) [inline api/models/Group.js:41]: `unique` and `index` are redundant with `primaryKey`

File: assets/app/app/serializer.js [hunk 6]
```diff
@@ -25,7 +25,4 @@
 import DS from 'ember-data';
 
 export default DS.JSONAPISerializer.extend({
-  attrs: {
```
Comment (willrnch) [inline assets/app/app/serializer.js:28]: the whole file can be removed

File: assets/app/machine/serializer.js [hunk 7]
```diff
@@ -25,7 +25,4 @@
 import DS from 'ember-data';
 
 export default DS.JSONAPISerializer.extend({
```
Comment (willrnch) [inline assets/app/machine/serializer.js]: the whole file can be removed

File: assets/app/user/serializer.js [hunk 8]
```diff
@@ -25,7 +25,4 @@
 import DS from 'ember-data';
 
 export default DS.JSONAPISerializer.extend({
```
Comment (willrnch) [inline assets/app/user/serializer.js]: the whole file can be removed

File: api/controllers/AppController.js (hunk 1 above)
Comment (dynamiccast) [inline api/controllers/AppController.js:74]: Fixed

File: api/controllers/NanocloudController.js [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+    var file = "";
+    if (req.url === "/") {
+      file = "index.html";
+      res.set('Content-Type', 'text/html; charset=utf-8');
+    } else {
+      file = url.parse(req.url).pathname;
+      res.set('Content-Type', guessType(file));
+    }
+
+    var emberApp = __dirname + '/../../assets/dist/' + file;
```
Comment (willrnch) [inline api/controllers/NanocloudController.js]: `path.join(__dirname, '/../../assets/dist/', file);` to make codeclimate happy

**Pull request merged**